specreduce/background.py

```diff
@@ -259,14 +278,12 @@ def sub_image(self, image=None):
         -------
         array with same shape as ``image``
         """
-        if image is None:
-            image = self.image
+        image = self._parse_image(image)
 
-        if isinstance(image, NDData):
-            # https://docs.astropy.org/en/stable/nddata/mixins/ndarithmetic.html
-            return image.subtract(self.bkg_image(image)*image.unit)
-        else:
-            return image - self.bkg_image(image)
+        # https://docs.astropy.org/en/stable/nddata/mixins/ndarithmetic.html
+        # (compare_wcs argument needed to avoid TypeError from SpectralCoord
+        #  when image's spectral axis is in pixels)
+        return image.subtract(self.bkg_image(image), compare_wcs=None)
```

Comment:
this line seems to be dropping the units on the spectral axis (if I pass `image` as a `Spectrum1D` with wavelength units, `sub_image()` is returning in pixels and have confirmed that both `image` and `self.bkg_image(image)` still have the correct wavelength units).  Removing `compare_wcs=None` fixes the problem, so we might need something like the following since `compare_wcs=None` is needed to support the pixels-case (feel free to refactor however makes sense to you):

```
from astropy import units as u
bkg_image = self.bkg_image(image)
kwargs = {'compare_wcs': None} if bkg_image.spectral_axis.unit == u.pix else {}
return image.subtract(self.bkg_image(image), **kwargs)
```

Also see https://github.com/astropy/specutils/issues/989 which was found while debugging this but can probably be ignored here with the assumption that specutils will properly handle the spectral axes comparison in the future.

Reply:
That makes sense. I got this working in the latest commit and also added tests that would have caught the situation you encountered here. I agree with deferring a spectral axis compatibility check to specutils.

Reply:
awesome, thanks!  Can confirm that it now works as expected for the wavelength-units case